authentication/authentication-common/src/main/java/io/mosip/authentication/common/service/impl/notification/NotificationServiceImpl.java

```diff
@@ -1,381 +1,403 @@
-		String otpSubjectTemplate = EnvUtil.getOtpSubjectTemplate();
-
-		String contentTemplate = "";
-		String subjectTemplate = "";
-		if (sender == SenderType.AUTH && authEmailSubjectTemplate != null && authEmailContentTemplate != null) {
-			subjectTemplate = authEmailSubjectTemplate;
-			contentTemplate = authEmailContentTemplate;
-		} else if (sender == SenderType.OTP && otpSubjectTemplate != null && otpContentTemaplate != null) {
-			subjectTemplate = otpSubjectTemplate;
-			contentTemplate = otpContentTemaplate;
-		}
-
-		String mailSubject = applyTemplate(values, subjectTemplate, templateLanguages);
-		String mailContent = applyTemplate(values, contentTemplate, templateLanguages);
-		notificationManager.sendEmailNotification(emailId, mailSubject, mailContent);
-	}
-	
-	/**
-	 * This method gets the template languages in following order.
-	 * 1. Gets user preferred languages if not
-	 * 2. Gets default template languages from configuration if not
-	 * 3. Gets the data capture languages
-	 * @param idInfo
-	 * @return
-	 * @throws IdAuthenticationBusinessException
-	 */
-	private List<String> getTemplateLanguages(Map<String, List<IdentityInfoDTO>> idInfo)
-			throws IdAuthenticationBusinessException {
-		List<String> userPreferredLangs = idInfoFetcher.getUserPreferredLanguages(idInfo);
-		List<String> defaultTemplateLanguges = userPreferredLangs.isEmpty()
-				? idInfoFetcher.getTemplatesDefaultLanguageCodes()
-				: userPreferredLangs;
-		if (defaultTemplateLanguges.isEmpty()) {
-			List<String> dataCaptureLanguages = languageUtil.getDataCapturedLanguages(DemoMatchType.NAME, idInfo);
-			Collections.sort(dataCaptureLanguages, languageComparator);
-			return dataCaptureLanguages;
-		}
-
-		return defaultTemplateLanguges;
-
-	}
+    /**
+     * The Constant AUTH_TYPE.
+     */
+    private static final String AUTH_TYPE = "authType";
+    /**
+     * The Constant NAME.
+     */
+    private static final String NAME = "name";
+    /**
+     * The Constant TIME.
+     */
+    private static final String TIME = "time";
+    /**
+     * The Constant DATE.
+     */
+    private static final String DATE = "date";
+
+    @Autowired
+    private IdInfoFetcher idInfoFetcher;
+
+    /**
+     * ID Template manager
+     */
+    @Autowired
+    private IdTemplateManager idTemplateManager;
+
+    @Autowired
+    private NotificationManager notificationManager;
+
+    @Autowired
+    @Qualifier("NotificationLangComparator")
+    private LanguageComparator languageComparator;
+
+    @Autowired
+    private EntityInfoUtil entityInfoUtil;
+
+    @Autowired
+    private LanguageUtil languageUtil;
+
+    @Value("${mosip.ida.otp.notification.request.mode:0}")  // 0 sync 1 async
+    private int requestMode;
+
+    private static final Logger logger = IdaLogger.getLogger(NotificationServiceImpl.class);
+
+    @Value("${mosip.ida.email.timeout:15}")
+    private long emailTimeout;
+
+    @Value("${mosip.ida.sms.timeout:15}")
+    private long smsTimeout;
```

Comment:
added in config ?

Reply:
https://github.com/mosip/mosip-config/pull/9642